tests/Unit/App/Http/Controllers/Admin/OrderControllerTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Tests\Unit\App\Http\Controllers\Admin;
+
+use Tests\TestCase;
+use Mockery;
+use App\Models\Order;
+use App\Models\OrderDetail;
+use App\Repositories\Order\OrderRepositoryInterface;
+use App\Http\Controllers\Admin\OrderController;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+
+class OrderControllerTest extends TestCase
+{
+    protected $mockOrderRepository, $controller;
+
+    public function setUp() : void
+    {
+        parent::setUp();
+        $this->mockOrderRepository = Mockery::mock(OrderRepositoryInterface::class)->makePartial();
+        $this->controller = new OrderController($this->mockOrderRepository);
+    }
+
+    public function tearDown() : void
+    {
+        Mockery::close();
+        unset($this->controller);
+        parent::tearDown();
+    }
+
+    public function test_index_return_view()
+    {
+        $view = $this->controller->index();
+        $this->assertEquals('admin.orders.index', $view->getName());
+    }
+
+    public function test_load_orders()
+    {
+        $orders = factory(Order::class, config('app.fake_records_number'))->raw();
+
+        $this->mockOrderRepository
+            ->shouldReceive('loadOrders')
+            ->andReturn($orders);
+        $result = $this->controller->loadOrders();
+
+        $this->assertSame($orders, $result);
+    }
+
+    public function test_view_details()
+    {
+        $id = config('app.default_record_id');
+        $orderDetails = factory(OrderDetail::class, config('app.fake_records_number'))->raw();
+        // dd($orderDetails);
+
+        $this->mockOrderRepository
+            ->shouldReceive('getDetails')
+            ->with($id)
+            ->andReturn($orderDetails);
+
+        $view = $this->controller->viewDetails($id);
+        //Kiem tra ten view tra ve
+        $this->assertEquals('admin.order-details.index', $view->getName());
+        //Kiem tra trong Array tra ve co Key nay khong
+        $this->assertArrayHasKey('orderDetails', $view->getData());
+    }
+
+    public function test_deny_order()
+    {
+        $request = new Request();
+        $request->orderId = config('app.default_record_id');
+        $order = factory(Order::class, 1)->make();
```

Comment:
```suggestion
        $order = factory(Order::class)->make();
```
Hàm find trả về một `Model` chứ ko phải một `Collection` đúng ko anh? vậy sao truyền số vào chi nhỉ o_O?

Reply:
phần hàm shouldReceive('find') thì a hiểu nó là giả lập hành động gọi đến hàm find() khi truyền thêm with($var) thì nó sẽ chỉ định tham số truyền vào. Chứ a nghĩ nó thực hiện xong thì mình mới quan tâm đến kết quả trả về.

Reply:
ý e ở đây là việc anh giả lập hàm chưa chính xác. Vì hàm này đúng lý sẽ là trả về một Model nhưng factory make thế kia nó sẽ ra một Collection với 1 giá trị. Thì có thể hiện tại method anh test đang đơn giản nó chưa sai nhưng giả sử sau này a làm Unit Test với một project lớn thì nó sẽ phức tạp hơn rất nhiều và khác biệt giữa model và một collection của model sẽ có thể gây lỗi